lib/WorkerHandler.js

```diff
@@ -127,12 +158,12 @@ function WorkerHandler(script, options) {
       // find the task from the processing queue, and run the tasks callback
       var id = response.id;
       var task = me.processing[id];
-      if (task) {
+      if (task !== undefined) {
         // remove the task from the queue
         delete me.processing[id];
 
         // test if we need to terminate
-        if (me.terminating) {
+        if (me.terminating === true) {
```

Comment:
Coercion here seemed to pop up in the profiler, so I simply made the check explicit.